### PR TITLE
🐛 Source Amazon Seller Partner: use replication end date from user provided config

### PR DESCRIPTION
**Pull request by ganpatagarwal**
## What
* Stream Name: GET_SALES_AND_TRAFFIC_REPORT
With current implementation of `stream_slices` method, `replication_end_date` config value provided by user is not being used. This is inflating the number of days for which data needs to be pulled.

## How
* Added a check for presence of valid `replication_end_date` and updating local `end_date` with `replication_end_date`

## Recommended reading order
1. `streams.py`

## 🚨 User Impact 🚨
No breaking change.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [x] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [x] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

Results (1.02s):
      94 passed
Name                                                   Stmts   Miss  Cover   Missing
------------------------------------------------------------------------------------
source_amazon_seller_partner/__init__.py                   2      0   100%
source_amazon_seller_partner/auth.py                      61     36    41%   24, 45, 48-99, 102-108
source_amazon_seller_partner/constants.py                 37      1    97%   67
source_amazon_seller_partner/source.py                    51     12    76%   134-151
source_amazon_seller_partner/spec.py                      23      0   100%
source_amazon_seller_partner/streams.py                  610    273    55%   62, 65, 68, 77, 82, 87, 92, 97, 102-114, 117-120, 128, 135-138, 191, 195, 198, 201, 204, 224-235, 244-261, 270-279, 282-290, 296-299, 304-309, 312, 315-318, 331-364, 465-468, 515-522, 535-539, 552-556, 597-601, 608-609, 618-622, 626-654, 704, 713-722, 729-732, 738-761, 811, 825-830, 858, 863-865, 870, 873-877, 895-898, 910, 919-928, 934-946, 953-956, 962-991, 1031, 1036-1047, 1050, 1063, 1068, 1103-1107, 1121, 1138, 1155-1162, 1192-1230
unit_tests/test_finance_streams.py                        63      1    98%   157
unit_tests/test_reports_streams_rate_limits.py            33      0   100%
unit_tests/test_reports_streams_settlement_report.py      23      0   100%
unit_tests/test_source.py                                 42      0   100%
unit_tests/test_transform_function.py                     23      0   100%
------------------------------------------------------------------------------------
TOTAL                                                    968    323    67%

</details>

### CODE REVIEW

**Comment by marcosmarxm** [comment]
Hello :wave:, first thank you for this amazing contribution. 

We really appreciate the effort you've made to improve the project.
We ask you patience for the code review. Last month our team was focused on Hacktoberfest event and that probably left some PR without the proper feedback. And this week, due to the Thanksgiving US Holiday, most our team is out of office with their families. Another important piece of information why code won't be merge this week is: as a safety measure the core team has decided to freeze merging code to main branch to keep the release stable. Next week we'll return to you with the proper code review and update the status of your contribution.

If you have any questions feel free to send me a message in Slack!
Thanks!

**Comment by geekwhocodes** [comment]
Hi @ganpatagarwal, thanks for your amazing contribution. 

`'order'` property is required for all fields defined in the `'spec.json'` file. Can you please update `'spec.json'` file?

**Comment by ganpatagarwal** [comment]
> Hi @ganpatagarwal, thanks for your amazing contribution.
> 
> `'order'` property is required for all the fields defined in `'spec.json'` file. Can you please update `'spec.json'` file?

I am not making any changes to `spec.json` file in my PR? Shouldn't that be done by the code maintainers?

**Comment by geekwhocodes** [comment]
> > Hi @ganpatagarwal, thanks for your amazing contribution.
> > `'order'` property is required for all the fields defined in `'spec.json'` file. Can you please update `'spec.json'` file?
> 
> I am not making any changes to `spec.json` file in my PR? Shouldn't that be done by the code maintainers?

I understand that you haven't touched `spec.json` file but we are all working towards following standards to meet minimum requirements. I request you to add `order` value for remaining properties in `spec.json` after than we will able to merge your changes. 

@sajarin any thoughts on this?

**Comment by geekwhocodes** [comment]
@ganpatagarwal 
I see some override's here - https://github.com/airbytehq/airbyte/blob/f57fcdf87cf2572758e1a3dbb3dac5faa93d6609/airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/source.py#L222 that's why generated spec is missing order values for those two fields. 
cc: @sajarin 

Can you remove `order` field for those two fields please? After that we will be good, also share acceptance tests screenshot.

**Comment by ganpatagarwal** [comment]
removed the order for them

```
python -m pytest -p integration_tests.acceptance
Test session starts (platform: darwin, Python 3.9.11, pytest 6.2.5, pytest-sugar 0.9.5)
cachedir: .pytest_cache
hypothesis profile 'default' -> database=DirectoryBasedExampleDatabase('/Users/ganpatagarwal/stoke/airbyte/airbyte-integrations/connectors/source-amazon-seller-partner/.hypothesis/examples')
rootdir: /Users/ganpatagarwal/stoke/airbyte, configfile: pytest.ini
plugins: sugar-0.9.5, requests-mock-1.9.3, mock-3.6.1, hypothesis-6.54.3, timeout-1.4.2, cov-3.0.0
collecting ... {"type": "LOG", "log": {"level": "WARN", "message": "The acceptance-test-config.yml file is in a legacy format. Please migrate to the latest format."}}
{"type": "LOG", "log": {"level": "WARN", "message": "The acceptance-test-config.yml file is in a legacy format. Please migrate to the latest format."}}
{"type": "LOG", "log": {"level": "WARN", "message": "The acceptance-test-config.yml file is in a legacy format. Please migrate to the latest format."}}
{"type": "LOG", "log": {"level": "WARN", "message": "The acceptance-test-config.yml file is in a legacy format. Please migrate to the latest format."}}
{"type": "LOG", "log": {"level": "WARN", "message": "The acceptance-test-config.yml file is in a legacy format. Please migrate to the latest format."}}
{"type": "LOG", "log": {"level": "WARN", "message": "The acceptance-test-config.yml file is in a legacy format. Please migrate to the latest format."}}
{"type": "LOG", "log": {"level": "WARN", "message": "The acceptance-test-config.yml file is in a legacy format. Please migrate to the latest format."}}
{"type": "LOG", "log": {"level": "WARN", "message": "The acceptance-test-config.yml file is in a legacy format. Please migrate to the latest format."}}
{"type": "LOG", "log": {"level": "WARN", "message": "The acceptance-test-config.yml file is in a legacy format. Please migrate to the latest format."}}
{"type": "LOG", "log": {"level": "WARN", "message": "The acceptance-test-config.yml file is in a legacy format. Please migrate to the latest format."}}
{"type": "LOG", "log": {"level": "WARN", "message": "The acceptance-test-config.yml file is in a legacy format. Please migrate to the latest format."}}
{"type": "LOG", "log": {"level": "WARN", "message": "The acceptance-test-config.yml file is in a legacy format. Please migrate to the latest format."}}
{"type": "LOG", "log": {"level": "WARN", "message": "The acceptance-test-config.yml file is in a legacy format. Please migrate to the latest format."}}
{"type": "LOG", "log": {"level": "WARN", "message": "The acceptance-test-config.yml file is in a legacy format. Please migrate to the latest format."}}
{"type": "LOG", "log": {"level": "WARN", "message": "The acceptance-test-config.yml file is in a legacy format. Please migrate to the latest format."}}
{"type": "LOG", "log": {"level": "WARN", "message": "The acceptance-test-config.yml file is in a legacy format. Please migrate to the latest format."}}
{"type": "LOG", "log": {"level": "WARN", "message": "The acceptance-test-config.yml file is in a legacy format. Please migrate to the latest format."}}
{"type": "LOG", "log": {"level": "WARN", "message": "The acceptance-test-config.yml file is in a legacy format. Please migrate to the latest format."}}
{"type": "LOG", "log": {"level": "WARN", "message": "The acceptance-test-config.yml file is in a legacy format. Please migrate to the latest format."}}
{"type": "LOG", "log": {"level": "WARN", "message": "The acceptance-test-config.yml file is in a legacy format. Please migrate to the latest format."}}
{"type": "LOG", "log": {"level": "WARN", "message": "The acceptance-test-config.yml file is in a legacy format. Please migrate to the latest format."}}
{"type": "LOG", "log": {"level": "WARN", "message": "The acceptance-test-config.yml file is in a legacy format. Please migrate to the latest format."}}
{"type": "LOG", "log": {"level": "WARN", "message": "The acceptance-test-config.yml file is in a legacy format. Please migrate to the latest format."}}
{"type": "LOG", "log": {"level": "WARN", "message": "The acceptance-test-config.yml file is in a legacy format. Please migrate to the latest format."}}
{"type": "LOG", "log": {"level": "WARN", "message": "The acceptance-test-config.yml file is in a legacy format. Please migrate to the latest format."}}
{"type": "LOG", "log": {"level": "WARN", "message": "The acceptance-test-config.yml file is in a legacy format. Please migrate to the latest format."}}
{"type": "LOG", "log": {"level": "WARN", "message": "The acceptance-test-config.yml file is in a legacy format. Please migrate to the latest format."}}
{"type": "LOG", "log": {"level": "WARN", "message": "The acceptance-test-config.yml file is in a legacy format. Please migrate to the latest format."}}
{"type": "LOG", "log": {"level": "INFO", "message": "Using /Users/ganpatagarwal/stoke/airbyte/airbyte-integrations/connectors/source-amazon-seller-partner/secrets/config.json as configuration. It is the most recent version."}}

 airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py::TestSpec.test_config_match_spec[inputs0] ✓                                         4% ▌         {"type": "LOG", "log": {"level": "INFO", "message": "Using /Users/ganpatagarwal/stoke/airbyte/airbyte-integrations/connectors/source-amazon-seller-partner/secrets/config.json as configuration. It is the most recent version."}}

 airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py::TestSpec.test_match_expected[inputs0] ✓                                            8% ▉         {"type": "LOG", "log": {"level": "INFO", "message": "Using /Users/ganpatagarwal/stoke/airbyte/airbyte-integrations/connectors/source-amazon-seller-partner/secrets/config.json as configuration. It is the most recent version."}}

 airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py::TestSpec.test_docker_env[inputs0] ✓                                               12% █▍        {"type": "LOG", "log": {"level": "INFO", "message": "Using /Users/ganpatagarwal/stoke/airbyte/airbyte-integrations/connectors/source-amazon-seller-partner/secrets/config.json as configuration. It is the most recent version."}}

 airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py::TestSpec.test_enum_usage[inputs0] ✓                                               17% █▋        {"type": "LOG", "log": {"level": "INFO", "message": "Using /Users/ganpatagarwal/stoke/airbyte/airbyte-integrations/connectors/source-amazon-seller-partner/secrets/config.json as configuration. It is the most recent version."}}

 airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py::TestSpec.test_oneof_usage[inputs0] ✓                                              21% ██▏
 airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py::TestSpec.test_required[inputs0] ✓                                                 25% ██▌
 airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py::TestSpec.test_optional[inputs0] ✓                                                 29% ██▉
 airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py::TestSpec.test_has_secret[inputs0] ✓                                               33% ███▍
 airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py::TestSpec.test_secret_never_in_the_output[inputs0] ✓                               38% ███▊      {"type": "LOG", "log": {"level": "INFO", "message": "Using /Users/ganpatagarwal/stoke/airbyte/airbyte-integrations/connectors/source-amazon-seller-partner/secrets/config.json as configuration. It is the most recent version."}}

 airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py::TestSpec.test_secret_is_properly_marked[inputs0] ✓                                42% ████▎     {"type": "LOG", "log": {"level": "INFO", "message": "Using /Users/ganpatagarwal/stoke/airbyte/airbyte-integrations/connectors/source-amazon-seller-partner/secrets/config.json as configuration. It is the most recent version."}}

 airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py::TestSpec.test_defined_refs_exist_in_json_spec_file[inputs0] ✓                     46% ████▋     {"type": "LOG", "log": {"level": "INFO", "message": "Using /Users/ganpatagarwal/stoke/airbyte/airbyte-integrations/connectors/source-amazon-seller-partner/secrets/config.json as configuration. It is the most recent version."}}

 airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py::TestSpec.test_oauth_flow_parameters[inputs0] ✓                                    50% █████     {"type": "LOG", "log": {"level": "INFO", "message": "Using /Users/ganpatagarwal/stoke/airbyte/airbyte-integrations/connectors/source-amazon-seller-partner/secrets/config.json as configuration. It is the most recent version."}}

 airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py::TestSpec.test_backward_compatibility[inputs0] ✓                                   54% █████▌    {"type": "LOG", "log": {"level": "INFO", "message": "Using /Users/ganpatagarwal/stoke/airbyte/airbyte-integrations/connectors/source-amazon-seller-partner/secrets/config.json as configuration. It is the most recent version."}}

 airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py::TestSpec.test_additional_properties_is_true[inputs0] ✓                            58% █████▉    {"type": "LOG", "log": {"level": "INFO", "message": "Using /Users/ganpatagarwal/stoke/airbyte/airbyte-integrations/connectors/source-amazon-seller-partner/integration_tests/invalid_config.json as configuration. It is the most recent version."}}

 airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py::TestConnection.test_check[inputs0] ✓                                              62% ██████▍   {"type": "LOG", "log": {"level": "INFO", "message": "Using /Users/ganpatagarwal/stoke/airbyte/airbyte-integrations/connectors/source-amazon-seller-partner/secrets/config.json as configuration. It is the most recent version."}}
```

File: airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/spec.py
Comment (geekwhocodes) [inline airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/spec.py:110]: We are not able to add order to `aws_environment` because it is being overridden in https://github.com/airbytehq/airbyte/blob/f57fcdf87cf2572758e1a3dbb3dac5faa93d6609/airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/source.py#L222.

File: airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/spec.py
Comment (geekwhocodes) [inline airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/spec.py:113]: We are not able to add order to `aws_environment` because it is being overridden in https://github.com/airbytehq/airbyte/blob/f57fcdf87cf2572758e1a3dbb3dac5faa93d6609/airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/source.py#L222.

**Review by geekwhocodes — APPROVED** [review]

**Review by sajarin — APPROVED** [review]

**Pull request merged**
